### PR TITLE
Open ai

### PR DESCRIPTION
**Pull request by rachfop**

### CODE REVIEW

**Review by sumo43 — APPROVED** [review]
LGTM

File: docs/serverless/workers/vllm/openai-compatibility.md
Comment (alpayariyak) [inline docs/serverless/workers/vllm/openai-compatibility.md:1]: There's no `model:tag` naming - `MODEL_NAME` is the hugginface repo of the model

For an Endpoint ID abc, the syntax in the tutorial will result in .../v2/$abc/openai/v1, so $ needs to be removed. Same for the tutorial in get-started if it's copied. Also a page for non-openai is missing

**Review by alpayariyak — CHANGES_REQUESTED** [review]
I don't have access to Vercel, so these were some comments based on the raw markdown

File: docs/serverless/endpoints/overview.md [hunk 1]
```diff
@@ -3,15 +3,17 @@ title: Overview
...
+RunPod Endpoints serve as the gateway to deploying and managing your Serverless Workers.
+These endpoints allow for flexible interaction with a variety of models, supporting both asynchronous and synchronous operations tailored to your computational needs.
+Whether you're processing large data sets, requiring immediate results, or scheduling tasks to run in the background, RunPod's API Endpoints provide the versatility and scalability essential for modern computing tasks.
 
-### Key Features
+### Key features
 
-- **Asynchronous and Synchronous Jobs:** Choose the execution mode that best fits your workflow, whether it's a task that runs in the background or one that delivers immediate results.
+- **Asynchronous and synchronous Jobs:** Choose the execution mode that best fits your workflow, whether it's a task that runs in the background or one that delivers immediate results.
```
Comment (AllysonRosenthal) [inline docs/serverless/endpoints/overview.md:12]: Should jobs also be lowercase?

File: docs/serverless/endpoints/overview.md [hunk 2]
```diff
@@ -20,14 +22,16 @@ You can find your API key in the [user settings section](https://www.runpod.io/c
 
 :::note
 
-**Privacy and Security:** RunPod prioritizes your data's privacy and security. Inputs and outputs are retained for a maximum of 30 minutes to protect your information.
+**Privacy and Security:** RunPod prioritizes your data's privacy and security.
```
Comment (AllysonRosenthal) [inline docs/serverless/endpoints/overview.md:25]: Why is security capitalized here?

File: docs/serverless/workers/vllm/environment-variables.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+| `MAX_MODEL_LENGTH`  | Model's maximum  | `int`                                                     | Maximum number of tokens for the engine to handle per request.                                                                                                                                                                                                                                            |
+| `BASE_PATH`         | `/runpod-volume` | `str`                                                     | Storage directory for Huggingface cache and model. Utilizes network storage if attached when pointed at `/runpod-volume`, which will have only one worker download the model once, which all workers will be able to load. If no network volume is present, creates a local directory within each worker. |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:41]: Huggingface -> Hugging Face 

Confusing description. Maybe something like:
When network storage is attached, uses a single Worker to download the model once for all Workers to load. If no network storage is present, creates a local directory within each Worker and downloads the model for each.

File: docs/serverless/workers/vllm/environment-variables.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+### System, GPU, and tensor parallelism (Multi-GPU) Settings
+
+These settings optimize hardware utilization and parallel processing capabilities, enhancing performance and efficiency for your vLLM Worker.
+
+| Name                           | Default | Type/Choices     | Description                                                                                                                         |
+| ------------------------------ | ------- | ---------------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| `GPU_MEMORY_UTILIZATION`       | `0.95`  | `float`          | Sets GPU VRAM utilization.                                                                                                          |
+| `MAX_PARALLEL_LOADING_WORKERS` | `None`  | `int`            | Load model sequentially in multiple batches, to avoid RAM OOM when using tensor parallel and large models.                          |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:67]: Load -> loads, remove comma

File: docs/serverless/workers/vllm/environment-variables.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+| `BLOCK_SIZE`                   | `16`    | `8`, `16`, `32`  | Token block size for contiguous chunks of tokens.                                                                                   |
+| `SWAP_SPACE`                   | `4`     | `int`            | CPU swap space size (GiB) per GPU.                                                                                                  |
+| `ENFORCE_EAGER`                | `0`     | boolean as `int` | Always use eager-mode PyTorch. If False(`0`), will use eager mode and CUDA graph in hybrid for maximal performance and flexibility. |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:70]: will use -> uses

File: docs/serverless/workers/vllm/environment-variables.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+| `MAX_CONTEXT_LEN_TO_CAPTURE`   | `8192`  | `int`            | Maximum context length covered by CUDA graphs. When a sequence has context length larger than this, we fall back to eager mode.     |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:71]: we fall -> ~we~ falls

File: docs/serverless/workers/vllm/environment-variables.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+| `DISABLE_CUSTOM_ALL_REDUCE`    | `0`     | `int`            | Enables or disables custom all reduce.                                                                                              |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:72]: is "custom all reduce" common knowledge?

File: docs/serverless/workers/vllm/environment-variables.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+| `DEFAULT_BATCH_SIZE_GROWTH_FACTOR`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | `3`     | `float`      | Growth factor for dynamic batch size.                                                                |
+
+The first request will have a batch size of `DEFAULT_MIN_BATCH_SIZE`, and each subsequent request will have a batch size of `previous_batch_size * DEFAULT_BATCH_SIZE_GROWTH_FACTOR`.
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:84]: will have -> uses x2

File: docs/serverless/workers/vllm/environment-variables.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+This will continue until the batch size reaches `DEFAULT_BATCH_SIZE`.
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:85]: will continue -> continues

File: docs/serverless/workers/vllm/environment-variables.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+For example, the default values, the batch sizes will be `1, 3, 9, 27, 50, 50, 50, ...`.
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:87]: using the default values

File: docs/serverless/workers/vllm/environment-variables.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+You can also specify this per request, with inputs `max_batch_size`, `min_batch_size`, and `batch_size_growth_factor`. 
+This has nothing to do with vLLM's internal batching, but rather the number of tokens sent in each HTTP request from the worker.
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:89]: worker -> Worker

File: docs/serverless/workers/vllm/environment-variables.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+### OpenAI settings
+
+Configure compatibility with OpenAI's API, including output formatting and model naming.
+
+| Name                                | Default     | Type/Choices     | Description                                                                                                                                                                       |
+| ----------------------------------- | ----------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `RAW_OPENAI_OUTPUT`                 | `1`         | boolean as `int` | Enables raw OpenAI SSE format string output when streaming. Required to be enabled (which it is by default) for OpenAI compatibility.                                             |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:97]: Required to be -> must be

File: docs/serverless/workers/vllm/environment-variables.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+| `OPENAI_SERVED_MODEL_NAME_OVERRIDE` | `None`      | `str`            | Overrides the name of the served model from model repo/path to specified name, which you will then be able to use the value for the `model` parameter when making OpenAI requests |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:98]: you will then be able to -> you can

Clarify whether you mean
use the value for -> use as the value for/use the value as 
or something else

File: docs/serverless/workers/vllm/environment-variables.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+| `OPENAI_RESPONSE_ROLE`              | `assistant` | `str`            | Role of the LLM's Response in OpenAI Chat Completions.                                                                                                                            |
+
+### Serverless settings
+
+Maximize the scalability and efficiency of your vLLM Worker in serverless environments with concurrency and logging settings.
+
+| Name                   | Default | Type/Choices     | Description                                                                                                                                                                |
+| ---------------------- | ------- | ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `MAX_CONCURRENCY`      | `300`   | `int`            | Max concurrent requests per worker. vLLM has an internal queue, so you don't have to worry about limiting by VRAM, this is for improving scaling/load balancing efficiency |
```
Comment (AllysonRosenthal) [inline docs/serverless/workers/vllm/environment-variables.md:107]: VRAM, this is for improving -> VRAM. This improves 

And add period at the end of the description.

**Review by AllysonRosenthal — COMMENTED** [review]
Partial review. I've gotten through changed files through vllm/get-started. The Getting Started page has comments in Vercel, the rest are here.

**Pull request merged**